huanyu-li/week4/mon/calculator.rb

```diff
@@ -0,0 +1,161 @@
+# https://gist.github.com/textchimp/c2736668175eba4cfe8802e2402cc74d
+
+# Calculator
+# Explanation
+# You will be building a calculator which can perform multiple arithmetic operations. Your program should allow the user to choose which operation to perform, enter the values on which the operation is performed, and output the result.
+# Specification:
+# A user should be given a menu of operations
+# A user should be able to choose from the menu
+# A user should be able to enter numbers to perform the operation on
+# A user should be shown the result
+# The menu should be displayed again, and process should continue until the user selects a quit option from the menu
+# Phase 1
+# Calculator functionality
+# Calculator should be able to do basic arithmetic (+,-, *, /)
+# Phase 2
+# Advanced Calculator functionality
+# Calculator should be able to do more complex arithmetic (exponents, square roots) - STFW
+# Bonus
+# Mortgage Calculator
+# Calculate the monthly required payment given the other variables as input (look up the necessary variables)
+
+# BMI Calculator
+# Calculate the body mass index (BMI) for an individual, given their height and weight
+
+# Trip Calculator
+# Calculate a trip time and cost given inputs for
+
+# distance
+# miles per gallon
+# price per gallon
+# speed in miles per hour
+
+def menu
+    puts 'Welcome to the calculator'
+    puts '1. Addition +'
+    puts '2. Subtraction -'
+    puts '3. Multiplication *'
+    puts '4. Division /'
+    puts '5. Advanced Calculator'
+    puts '  5.1. Exponents'
+    puts '  5.2. Square Roots'
+    puts '  5.3. Mortgage Calculator'
+    puts '  5.4. BMI Calculator'
+    puts '  5.5. Trip Calculator'
+    puts '0. Exit'
+    print 'Please select a number: '
+    gets.to_i
+end
+
+
+
+def basic_calculator (selection)
+
+    case selection
+    when 1
+        puts 'Addition: a + b = c'
+        print "Please enter 'a' and 'b' values separated by whitespace: " 
```

Comment:
you can make this clearer and more DRY by having them call a method that just gets two nums - something similar to the below:

         def getNumber
                print "Enter the first number:"
                first = gets.chomp.to_i
                
                print "Enter the second number:"
                second = gets.chomp.to_i
         end